R/tags.R

```diff
@@ -921,10 +1031,26 @@ withTags <- function(code) {
 # Make sure any objects in the tree that can be converted to tags, have been
 tagify <- function(x) {
   rewriteTags(x, function(uiObj) {
-    if (isTag(uiObj) || isTagList(uiObj) || is.character(uiObj))
+    if (isTag(uiObj)) {
+      renderFn <- uiObj$.render
+      if (is.function(renderFn)) {
+        # Set render method to `NULL` to avoid confusion
+        uiObj$.render <- NULL
+        return(
+          tagify(renderFn(uiObj))
+        )
+      } else {
+        return(uiObj)
+      }
+    }
```

Comment:
I think this should happen in an `as.tags` method (similar to `as.tags.shiny.tag.function`)? That way if we want to add `.render` to `tagList()`, doing so will be simpler (and we can keep the implementation of `tagify()` 'simple')